src/CanvasRenderingContext2d.cc

```diff
@@ -389,6 +389,7 @@ Context2d::fill(bool preserve) {
       cairo_set_source(_context, new_pattern);
       cairo_pattern_destroy(new_pattern);
     } else {
+      cairo_pattern_set_filter(state->fillPattern, state->patternQuality);
```

Comment:
patterQuality isn't really a canvas feature.
I think, but i may be wrong, it is used to simulate imageSmoothingEnabled property.

In canvas pattern quality is always best, there is no other option.

In case this change is ok, then it should be applied also to the stroke part.

Reply:
Now applied to the stroke as well